cli/init.go

```diff
@@ -0,0 +1,451 @@
+	case "windows":
+		configDir, err = os.UserConfigDir()
+		if err != nil {
+			return "", err
+		}
+	case "darwin":
+		configDir, err = os.UserConfigDir()
+		if err != nil {
+			return "", err
+		}
+	default: // "linux" and other Unix-like systems
+		configDir = os.Getenv("XDG_CONFIG_HOME")
+		if configDir == "" {
+			configDir, err = os.UserConfigDir()
+			if err != nil {
+				return "", err
+			}
+		}
+	}
+
+	appDataDir := filepath.Join(configDir, appName)
+	err = os.MkdirAll(appDataDir, 0755)
+	if err != nil {
+		return "", err
+	}
+
+	return appDataDir, nil
+}
+
+func getLatestCommitSHA(url string) (string, error) {
+	resp, err := http.Get(url)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("failed to get latest commit SHA: %s", resp.Status)
+	}
+
+	var result struct {
+		Commit struct {
+			SHA string `json:"sha"`
+		} `json:"commit"`
+	}
+	err = json.NewDecoder(resp.Body).Decode(&result)
+	if err != nil {
+		return "", err
+	}
+
+	return result.Commit.SHA, nil
+}
+
+func readDirectories(path string) ([]string, error) {
+	files, err := os.ReadDir(path)
+	if err != nil {
+		return nil, err
+	}
+
+	var directories []string
+	for _, file := range files {
+		if file.IsDir() {
+			directories = append(directories, file.Name())
+		}
+	}
+
+	return directories, nil
+}
+
+func cleanDirectory(dir string) error {
+	d, err := os.Open(dir)
+	if err != nil {
+		return err
+	}
+	defer d.Close()
+
+	names, err := d.Readdirnames(-1)
+	if err != nil {
+		return err
+	}
+
+	for _, name := range names {
+		err = os.RemoveAll(filepath.Join(dir, name))
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func copyDir(src string, dst string) error {
+	return filepath.Walk(src, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		// Construct the destination path
+		relPath, err := filepath.Rel(src, path)
+		if err != nil {
+			return err
+		}
+		dstPath := filepath.Join(dst, relPath)
+
+		if info.IsDir() {
+			// Create the directory
+			if err := os.MkdirAll(dstPath, info.Mode()); err != nil {
+				return err
+			}
+		} else {
+			// Copy the file
+			if err := copyFile(path, dstPath); err != nil {
+				return err
+			}
+		}
+		return nil
+	})
+}
+
+func copyFile(srcFile string, dstFile string) error {
+	src, err := os.Open(srcFile)
+	if err != nil {
+		return err
+	}
+	defer src.Close()
+
+	dst, err := os.Create(dstFile)
+	if err != nil {
+		return err
+	}
+	defer dst.Close()
+
+	if _, err := io.Copy(dst, src); err != nil {
+		return err
+	}
+
+	// Copy file permissions
+	srcInfo, err := os.Stat(srcFile)
+	if err != nil {
+		return err
+	}
+	return os.Chmod(dstFile, srcInfo.Mode())
+}
+
+func initCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "init <template> [path]",
+		Short:   "Initialize a new Dispatch project",
+		GroupID: "dispatch",
+		RunE: func(cmd *cobra.Command, args []string) error {
```

Comment:
Moving this closure to a dedicated function would reduce the amount of whitespace on the left, it would help make the code more readable to have indentation start closer to the left side of the view:
```go
func initRunE(cmd *cobra.Command, args []string) error {
  ...
}
```

Reply:
I like it. GitHub's default indentation is annoying.